translator/lib/SPIRV/SPIRVReader.cpp

```diff
@@ -6510,6 +6538,31 @@ Function *SPIRVToLLVM::transFunction(SPIRVFunction *BF) {
       transValue(BInst, F, BB, false);
     }
   }
+
+  // Update phi nodes -- add missing incoming arcs.
+  // This is necessary because LLVM's CFG is a multigraph, while SPIR-V's
+  // CFG is not.
+  for (BasicBlock &BB : *F) {
+    DenseMap<BasicBlock*, unsigned> PredecessorsToAdd =
+      getBlockPredecessorCounts(&BB);
+
+    if (PredecessorsToAdd.empty())
+        continue;
+
+    // Add missing incoming arcs to each phi node that requires fixups.
+    for (PHINode &Phi : BB.phis()) {
+      const unsigned InitialNumIncoming = Phi.getNumIncomingValues();
+      for (unsigned i = 0; i != InitialNumIncoming; ++i) {
+        BasicBlock* Predecessor = Phi.getIncomingBlock(i);
+        Value* IncomingValue = Phi.getIncomingValue(i);
+
+        const unsigned NumMissingArcs = PredecessorsToAdd[Predecessor];
+        for (unsigned j = 1; j < NumMissingArcs; ++j)
+          Phi.addIncoming(IncomingValue, Predecessor);
+      }
+    }
+  }
+
```

Comment:
please clear BlockToPredecessorCount at the begin or end of function transFunction